modules/costmodel.py

```diff
@@ -0,0 +1,149 @@
+        result = 0.951 * np.exp(-0.002 * total_mileage / 1000) # Share or original value
+        return result
+    
+    def calculate_cost(self, veh, s_annual, con, cbat_spec, e_bat, t_bat):
+        
+        # Powertrain cost
+        c_pt_residual = (self.c_pt[veh]*self.residualValue(s_annual*self.servicelife)
+                         *self.r**-self.servicelife) #residual value
+        c_pt = self.c_pt[veh] - c_pt_residual #total powertrain cost
+        
+        # Taxes (due once a year)        
+        c_tax = sum([self.c_tax[veh]*self.r**-t 
+                      for t in range(1, self.servicelife+1)]) 
+        
+        # Maintenance costs (due twice a year)      
+        c_maint = sum([self.c_maintenance[veh]*s_annual/2*self.r**(-t/2)
+                       for t in range(0,2*self.servicelife)])
+        
+        # Toll costs (due weekly)      
+        c_toll = sum([self.z_toll*self.c_toll[veh]*s_annual/52*self.r**(-t/52)
+                     for t in range(0,52*self.servicelife)])
+        
+        # Energy consumption costs (due weekly)
+        c_ene = sum([self.c_ene[veh]*con*s_annual/52*self.r**(-t/52) 
+                     for t in range(0, 52*self.servicelife)])
+        
+        ## Battery Costs
+        if veh=="bet": # only for BET
+            
+            # Initial battery costs
+            c_bat_init = cbat_spec*self.c_cell2system*e_bat
+            
+            # Battery replacement costs
+            n_replacements = int(self.servicelife/t_bat) # number of replacements
+            t_replacement = [t_bat*n for n in range(1,n_replacements+1)] # time of replacments in years            
+            c_bat_replacements = [cbat_spec*self.c_cell2system*e_bat*self.r**-t for t in t_replacement]
+            c_bat_replacement = sum(c_bat_replacements)
+            
+            # Residual value
+            c_bat_scrappage = [self.z_scr*cbat_spec*self.c_cell2system*e_bat*self.r**-t for t in t_replacement] #scrappage values
+            bat_eol_soh = (1-self.servicelife/t_bat%1) #remaining capacity at end of lifetime
+            c_bat_eol = ((self.z_scr + bat_eol_soh*(1-self.z_scr))
+                         *cbat_spec*self.c_cell2system*e_bat* self.r**-self.servicelife) # Add resale value of remaining capacity at EOL
+            c_bat_residuals = c_bat_scrappage + [c_bat_eol]
+            c_bat_residual = -sum(c_bat_residuals)
+            
+            # Imputed interest
+            t_operation = [t_bat]*n_replacements + [self.servicelife - t_bat*n_replacements]#Vector of investment durations
+            c_bat_inv = [c_bat_init] + c_bat_replacements
+            avg_bound_investment = [(c_purchase+c_res)/2 for (c_purchase,c_res) 
+                                    in zip(c_bat_inv, c_bat_residuals)] #investment till scrappage
+            c_bat_imputed_interest = sum([invest*(self.r**t-1) for (invest,t)
+                                       in zip(avg_bound_investment, t_operation)])
```

Comment:
It's better to put this into a separate function, this makes the code more readable and also makes it clear which variables are only intermediate results and which ones you actually want to keep.